### DISCUSSION TITLE
--outputjson and --watch

### DISCUSSION THREAD

**Discussion by elbaro** [microsoft/pyright#2966] [Q&A]
I created an app that pretty-prints the pyright output. It uses the span information from --outputjson.

```sh
 ▷  1  Error: [pyright] reportGeneralTypeIssues
       ╭─[/home/user/black/src/black/__init__.py:42:25]
       │
    42 │ from black.nodes import is_string_token
       ·                         ───────┬───────  
       ·                                ╰───────── "is_string_token" is unknown import symbol
   ────╯
```

Now to add a watch feature to my app, I tried `--outputjson --watch` but it's not supported.
I could parse `--watch` but it does not provide the span information.
Another solution is to watch files myself and rerun pyright for the entire files (rerunning pyright against only changed files is not enough) which is inefficient.

Is there any way to use a lsp-mode outside vscode? (no other lsp feature than watch is needed)
Do you have a plan to support --watch with a machine readable output or span information?

**Comment by erictraut** [comment]
You're correct. The `--watch` feature doesn't currently work with `--outputjson`. This is an odd combination, and I didn't anticipate that anyone would want to use those features together. It's easy to make this combination work, so I've updated the code and done some quick manual tests to make sure it looks OK. This will be included in the next release.
